Unstructured-IO/unstructured
# chore: build with registry as cache

* updates docker-build.sh to use the image registry as a cache
* updates make commands to no longer explicitly set platform (but optionally can be assigned via env var)


## Testing

Previous CI docker-test run time: [33m40s](https://github.com/Unstructured-IO/unstructured/actions/runs/4622497100/jobs/8175291101)
New CI docker-test run time: [5m30s](https://github.com/Unstructured-IO/unstructured/actions/runs/4623319897/jobs/8177044494)

- Run `make docker-build` and note that most of steps will say CACHED.
- Run `make docker-build` on amd vs arm and use `docker inspect image unstructured:dev | grep Architecture` and note that the respective image for your platform was built
